Bug fix

Routine030 opened the pull request:
Fixed first review bugs.

Routine030 committed:
Fix issue 1, add config, mod .ignore and readme.

Routine030 committed:
Fix issue 2, separate middleware from route.

Routine030 committed:
Fix issue 4, modify test src path.

Routine030 committed:
Fix issue 3, remove post.js,
change model define in todo.js, bootstrap in www.

Routine030 committed:
Fix issue 5, create DB entry before run unit test,
it'll avoid error if DB entry does not exist.

smlsunxie commented:
route 定義檔含了 middleware 及 controller 部分，請拆出去，並且用 require 的方式載入

改善 route 的可讀性及整體程式碼簡潔

https://github.com/Routine030/todo_list/blob/bugFix/routes/index.js 


其中 controller 部分如這個區塊

https://github.com/Routine030/todo_list/blob/bugFix/routes/index.js#L63-L77

其他 api 請一併調整，route 定義檔 應該只有

`router.route('/').post(preCheckContent ,userController.create);` 

這樣的設置

smlsunxie commented:
前端部份的搜尋及日期篩選目前都是直接用前端的資料本地進行

請調整為用 api 進行搜尋及篩選，才能了解這邊對於前後端互動部份的相關知識及熟悉度

smlsunxie commented:
請加上 lint 目前的程式碼有些規範不符合規定，參考

https://larrylu.blog/improve-code-quality-using-eslint-742cf1f384f1

smlsunxie commented:
所有函式請用 await/async 來控制 flow，包含 test case

Routine030 committed:
Separate controller and middleware from route.

Routine030 commented:
針對前兩點已經修改完畢，但我不確定接下來的動作是要先把這份pr merge進去，我再繼續commit修改到branch，再開一個pr，還是我就直接繼續commit，直接再開下一個pr?

smlsunxie commented:
可以繼續在這個  pr 往下完成

Routine030 committed:
Fix middleware bug.

Routine030 committed:
Modified query method: using query database to get entries.

Routine030 committed:
Add unit test for query API.

Routine030 committed:
(1)Fixed ESLint err in partial backend files
(2)Attached ESLint config file

Routine030 commented:
第三、四項要求還未完成，今晚會補上

Routine030 committed:
Using async/await to refine code flow(backend and unit test)

Routine030 committed:
Using async/await to refine code flow(frontend)

smlsunxie commented on client/src/App.js:
```diff
@@ -26,61 +26,63 @@ class App extends React.Component{
+          ret = await $.ajax({
+                  method: "GET",
+                  url: "http://localhost:3001/todo?text="+searchValue,
                 })
```
縮排有點怪怪的，這邊你有加上 lint 的話，理論上 lint 會告訴你程式碼沒有依規定的地方

若用 vscode 的話，參考 https://wcc723.github.io/tool/2017/11/09/coding-style/

smlsunxie commented on client/src/App.js:
```diff
@@ -26,61 +26,63 @@ class App extends React.Component{
+
       searchBar.content = searchValue;
       this.setState({searchData: searchBar})
       if(searchValue===''){
```
這邊也會有 lint 警告才對

`if(searchValue===''){` 會要求 `if(searchValue === ''){`

smlsunxie commented on client/src/App.js:
```diff
@@ -26,61 +26,63 @@ class App extends React.Component{
+  async changeState(event){
       var searchValue=event.target.value;
-    var searchBar = this.state.searchData;
+      var searchBar = this.state.searchData;
```
這邊 `=` 附近的空白不一至

smlsunxie commented on client/src/App.js:
```diff
@@ -26,61 +26,63 @@ class App extends React.Component{
+        try{
+          ret = await $.ajax({
+                  method: "GET",
+                  url: "http://localhost:3001/todo?text="+searchValue,
```
url `http://localhost:3001` 請參數化處理

smlsunxie commented on client/src/App.js:
```diff
@@ -26,61 +26,63 @@ class App extends React.Component{
+        this.handleSearchTodoItem();
       }
   }
   handleSearchList(arr,type){
```
type 參數沒用到？

handleSearchList 主要是處理什麼事情？

Routine030 replied:
handleSearchList會將輸入的搜尋條件依type填入（可能是建立日期或是todolist內容），
之後再統一由handleSearchTodoItem顯示結果

smlsunxie commented on client/src/InputBlock.js:
```diff
@@ -21,31 +21,32 @@ class InputBlock extends React.Component{
     this.setState({createtodo:event.target.value})
   }    
-  submitTodo(){
+async submitTodo(){
```
submitTodo 指的是 createTodo?

這邊可讀性要再加強， submit 是比較抽象的詞，此函式應該有他主要的功能

Routine030 committed:
1.Modified init db method
2.Fixed ESLint err in partial frontend files
3.Removed unused file: InputTask.js

Routine030 committed:
1. Add parameter for URL
2. Add some comments
3. Modify function name: submitTodo -> createTodo

Routine030 commented:
再度作了以下修正：
1. Frontend ESlint error fix.
2. URL參數化處理
3. 加強可讀性（function name修改、增加註解）
